scalac-scoverage-runtime/shared/src/main/scala/scoverage/Invoker.scala

```diff
@@ -59,6 +66,80 @@ object Invoker {
     }
   }
 
+  /**
+   * We record that the given id has been invoked by appending its id to the coverage
+   * data file.
+   *
+   * This will happen concurrently on as many threads as the application is using,
+   * so we use one file per thread, named for the thread id.
+   *
+   * This method is not thread-safe if the threads are in different JVMs, because
+   * the thread IDs may collide.
+   * You may not use `scoverage` on multiple processes in parallel without risking
+   * corruption of the measurement file.
+   *
+   * Same method as above but added additional features to make it work with
+   * https://github.com/scoverage/scalac-scoverage-plugin/issues/265. Typically, we clean the dataDir
+   * and add coverage instruments to the measurement file at runtime rather than compile time. This is
+   * needed as ahe runtime process may be running on a different host than the compile time one.
+   * However, since the instruments are on the classpath, we can retrieve the instruments from different hosts.
+   *
+   * @param id the id of the statement that was invoked
+   * @param instrumentsDir the directory where the instrument data is held and needs to be copied from
+   */
+  def invokedUseEnvironment(id: Int, instrumentsDir: String): Unit = {
+    // Generating a hash to add as a subdir to the basedir. This is because every
+    // instrumented binary has a coverage object for which measurements will be generated
+    // and it is possible that a test file tests multiple instrumented binaries. Thus, to
+    // segregate reports for each source file, we need this hash. This was not an issue with
+    // previous version as then we explicitly specified a different [dataDir] each time.
+    val instrumentHash = md5HashString(instrumentsDir)
```

Comment:
Wouldn't the md5HashString be the same for the instrumentsDir anyways?

Reply:
Only test statements for the same source file will have the same hash. Consider a test file testing code from 3 source files, thus, we'll have 3 instrumented binaries. If we do not have the hash, then all the measurements will go to the parent dir specified by `SCOVERAGE_MEASUREMENT_PATH` generating a single measurement file which could lead to skewed reports. The hash will segregate measurements for all the source files generating 3 subdirs with the respective instruments and measurements for the 3 source files.

Reply:
An alternative would be to create a subdir containing the entire filepath in its name instead of the md5 hash. My only concern with that was filepath being too long.
I am updating the name to be the filepath(look at the `safe_name` method) for now. Please let me know if there are better suggestions?